pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -3925,6 +3925,132 @@ func TestReconcilePipeline_TaskSpecMetadata(t *testing.T) {
 	}
 }
 
+func TestReconcileWithTaskResultsInFinalTasks(t *testing.T) {
```

Comment:
I believe we are trying to deprecate test builders from our tests. :sweat_smile: https://github.com/tektoncd/pipeline/issues/3178